### PR TITLE
Fix deps of skipped ports getting added to the CI run

### PR DESCRIPTION
**Pull request by Neumann-A**
Provided by Claude Sonnet 4.5 after longer discussions. 

What I like:
- ~~FilteredPortFileProvider~~ (damn was unnecessary)
- refactor

What I don like:
- Could be a bit less verbose. 
- probably I could fine tune it more.
- renaming stuff was a bit slow

alternative to #1822

### CODE REVIEW

File: src/vcpkg/commands.ci.cpp [hunk 1]
```diff
@@ -400,6 +430,18 @@ namespace vcpkg
         LocalizedString not_supported_regressions;
         {
             std::string msg;
+            
+            // First, print ports that were excluded early (before dependency resolution)
+            for (const auto& excluded_spec : ports_to_exclude)
```
Comment (BillyONeal) [inline src/vcpkg/commands.ci.cpp:446]: This is a good idea I missed in my version of the same https://github.com/microsoft/vcpkg-tool/pull/1822#issuecomment-3429660695

File: src/vcpkg/commands.ci.cpp [hunk 2]
```diff
@@ -400,7 +439,19 @@ namespace vcpkg
...
         {
             std::string msg;
-            for (const auto& spec : all_default_full_specs)
+
+            // First, print ports that were excluded early (before dependency resolution)
+            for (const auto& excluded_spec : ports_to_exclude)
+            {
+                split_specs->known.emplace(excluded_spec, BuildResult::Excluded);
+                // Generate a simple hash for excluded ports (they don't get real ABIs)
+                std::string fake_abi = "0000000000000000000000000000000000000000000000000000000000000000";
```
Comment (Thomas1664) [inline src/vcpkg/commands.ci.cpp]: This looks suboptimal because the string is copied anyways so we don't need to regenerate the same string at each iteration. And at this point it can also be converted to a constexpr string literal.

**Comment by Neumann-A** [comment]
should this be closed in favor of https://github.com/microsoft/vcpkg-tool/pull/1822 ?

**Comment by BillyONeal** [comment]
> should this be closed in favor of #1822 ?

Probably not, since I used ideas in here and am still looking at it a bit. Hope to have that ready for review with more tests and the edge cases the team asked me to investigate tomorrow.

File: src/vcpkg/commands.ci.cpp (hunk 1 above)
Comment (BillyONeal) [reply src/vcpkg/commands.ci.cpp:446]: In https://github.com/microsoft/vcpkg-tool/pull/1822#issuecomment-3429657274 in addition to fixing this problem I also fixed the "Summary" at the end to properly note that these were `EXCLUDED`

**Comment by BillyONeal** [comment]
Closing this as its effect was merged into #1822
